client_plugin/drivers/shared/etcdops.go

```diff
@@ -269,8 +305,122 @@ func etcdService(cmd []string) {
 	}
 }
 
-func (d *VolumeDriver) createEtcdClient() *etcdClient.Client {
-	dclient := d.dockerd
+// checkLocalEtcd function check if local ETCD endpoint is successfully started or not
+// if yes, start the watcher for volume global refcount
+func (e *etcdKVS) checkLocalEtcd() error {
+	ticker := time.NewTicker(checkSleepDuration)
+	defer ticker.Stop()
+	timer := time.NewTimer(requestTimeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			log.Infof("Checking etcd client is started")
+			cli, err := addrToEtcdClient(e.nodeAddr)
+			if err != nil {
+				log.WithFields(
+					log.Fields{"nodeAddr": e.nodeAddr,
+						"error": err},
+				).Warningf("Failed to get etcd client, retry before timeout ")
+			} else {
+				e.client = cli
+				go e.etcdWatcher()
+				return nil
+			}
+		case <-timer.C:
+			return fmt.Errorf("Timeout reached; etcd cluster is not started")
+		}
+	}
+}
+
+// etcdWatcher function sets up a watcher to monitor all the changes to global refcounts in the KV store
+func (e *etcdKVS) etcdWatcher() {
+	// TODO: when the manager is demoted to worker, the watcher should be cancelled
+	watchCh := e.client.Watch(context.Background(), etcdPrefixGRef,
+		etcdClient.WithPrefix(), etcdClient.WithPrevKV())
+	for wresp := range watchCh {
+		for _, ev := range wresp.Events {
+			e.etcdEventHandler(ev)
+		}
+	}
+}
+
+// etcdEventHandler function handles the returned event from etcd watcher of global refcount changes
+func (e *etcdKVS) etcdEventHandler(ev *etcdClient.Event) {
+	log.WithFields(
+		log.Fields{"type": ev.Type},
+	).Infof("Watcher on global refcount returns event ")
+
+	nested := func(key string, fromState volStatus, toState volStatus, fn func(string) bool) {
+		// watcher observes global refcount critical change
+		// transactional edit state first
+		volName := strings.TrimPrefix(key, etcdPrefixGRef)
+		succeeded := e.CompareAndPut(etcdPrefixState+volName,
+			string(fromState), string(volStateIntermediate))
+		if !succeeded {
+			// this handler doesn't get the right to start server
+			return
+		}
+
+		if fn(volName) {
+			// server start/stop succeed, set desired state
+			if e.CompareAndPut(etcdPrefixState+volName,
+				string(volStateIntermediate),
+				string(toState)) == false {
+				// Failed to set state to desired state
+				// set to state Error
+				e.CompareAndPut(etcdPrefixState+volName,
+					string(volStateIntermediate),
+					string(volStateError))
```

Comment:
Yes, users can still delete the volume with "--force" flag. And we already have the log message inside CompareAndPut function.